DigitalLearningSolutions.Data/DataServices/UserDataService/UserCentreDetailsDataService.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace DigitalLearningSolutions.Data.DataServices.UserDataService
+{
+    using System.Collections.Generic;
+    using System.Data;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Exceptions;
+
+    public partial class UserDataService
+    {
+        public void CreateOrUpdateUserCentreDetails(
+            int userId,
+            int centreId,
+            string email,
+            IDbTransaction? transaction = null

```

Comment:
If this is null, we should open our own transaction so that the locks we take out when checking if an email is already present are held until we finish adding the new email.